### DISCUSSION TITLE
Updating Cluster id when new records are added.

### DISCUSSION THREAD

**Discussion by AJAYRapidops** [moj-analytical-services/splink#875] [General]
Hello,
When i train a model with let say 'N' records and have saved the model and linker.predict().
Now when i add let say 'M records to the previous "N" records and initialise the linker object with the previous trained model and retrieve the linker.predict table and do linker.cluster_pairwise_predictions_at_threshold(), it generates new clusters even for the records which are exact match from the "N" records.
Do i have to train the model again and do linker.predict every time when the new records gets added to the database?

**Comment by RobinL** [comment]
You have to run `linker.predict()`, but you don't have to retrain the model.  You can save and load a model using the [load](https://moj-analytical-services.github.io/splink/linker.html#splink.linker.Linker.load_settings_from_json) and [save](https://moj-analytical-services.github.io/splink/linker.html#splink.linker.Linker.save_settings_to_json) functionality.

See also https://github.com/moj-analytical-services/splink/discussions/869#discussioncomment-4075019
